### PR TITLE
Updated Readme.md of Calendar Carousel

### PR DESCRIPTION
**Pull request by shljessie**
## Description

_Review of the Calendar Carousel dooboo-ui component. I will look over 
- failed yarn test sections  = **SOLVED**
- smooth scrollview errors
- clean code
- revision of README.md 
_

## Related Issues

_ 1) Slow Scrollview Rendering
        : Currently, the dates in the calendar renders slowly . It looks like the image below.
slow scrollview

       
The code for this section currently looks like this. 
Screen Shot 2020-08-19 at 1 45 58 PM

scrollEffect is the function that gets called each time someone scrolls
It sets the currentDate to the previous month or the next month depending on the position after the scroll
Screen Shot 2020-08-19 at 1 47 05 PM

renderCalendars does this.
Screen Shot 2020-08-19 at 1 46 29 PM

Any ideas on how I can fix this?

   2) Getting this error when I run yarn test  **SOLVED**
    
Screen Shot 2020-08-19 at 1 36 07 PM

--> This is solved ! Seems that I declared the markedDayEvents props wrong
._

## Tests

I added the following tests:

_Replace this with a list of the tests that you added as part of this PR. A change in behavior with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage](https://codecov.io/gh/dooboolab/dooboo-ui)._

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by shljessie** [comment]
Pushed the updated version of the readMe.md in Calendar Carousel

Currently, yarn test fails at

image

File: packages/CalendarCarousel/README.md [hunk 1]
```diff
@@ -64,6 +77,75 @@ import { CalendarCarousel } from 'dooboo-ui/lib/CalendarCarousel';
...
+        horizontal
+        pagingEnabled
+        scrollEventThrottle={16}
+        contentOffset = {{ x: layoutWidth, y: 0 }}
+        ref={scrollRef}
+        onMomentumScrollEnd={scrollEffect}
+      >
+        {renderCalendars(currentDate)}
+      </ScrollView>
+    </SafeAreaView>
```
Comment (hyochan) [inline packages/CalendarCarousel/README.md]: Please fix their indentation

File: packages/CalendarCarousel/README.md [hunk 2]
```diff
@@ -64,6 +77,75 @@ import { CalendarCarousel } from 'dooboo-ui/lib/CalendarCarousel';
...
+    const prevMonth = new Date(currentDate.getFullYear(), currentDate.getMonth() - 1, currentDate.getDate());
+    const currentMonth = new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate());
+    const nextMonth = new Date(currentDate.getFullYear(), currentDate.getMonth() + 1, currentDate.getDate());
+    return <Fragment>
+      {renderCalendar(prevMonth)}
+      {renderCalendar(currentMonth)}
+      {renderCalendar(nextMonth)}
+    </Fragment>;
+  };
+```	```
```
Comment (hyochan) [inline packages/CalendarCarousel/README.md]: Two ```

File: packages/CalendarCarousel/README.md [hunk 3]
```diff
@@ -64,6 +77,75 @@ import { CalendarCarousel } from 'dooboo-ui/lib/CalendarCarousel';
...
+        <View style={styles.weekdayStyle}>{weekdays}</View>
+        <FlatList
+          style={styles.dayContainer}
+          data={calendarDates}
+          numColumns={7}
+          renderItem ={({ item }): ReactElement => renderDates(item)}
+          keyExtractor={(item, id): string => id.toString()}
+        />
+        {renderDayEvents()}
+      </View>
```
Comment (hyochan) [inline packages/CalendarCarousel/README.md]: Move them left. There are too many empty spaces.

Comment (shljessie) [reply packages/CalendarCarousel/README.md]: fixed!

File: packages/CalendarCarousel/index.tsx [hunk 4]
```diff
@@ -178,49 +178,42 @@ function CalendarCarousel<T>({
...
   const month = currentDate.getMonth();
-  const [layoutWidth, setLayoutWidth] = useState<number>(330);
 
-  const isToday = (date: Date): boolean => {
+  const isToday = (dateitem: Date): boolean => {
     const today = new Date();
-    return date.getDate() === today.getDate() &&
-        date.getMonth() === today.getMonth() &&
-        date.getFullYear() === today.getFullYear();
+    return dateitem.getDate() === today.getDate() &&
```
Comment (hyochan) [inline packages/CalendarCarousel/index.tsx]: `dateitem` => `dateItem`

File: packages/CalendarCarousel/index.tsx [hunk 5]
```diff
@@ -290,16 +272,30 @@ function CalendarCarousel<T>({
...
+            <Text style = {styles.eventText}>{markedDayEvents[i].events}</Text>
+          </View>
+        );
+      }
+    });
+  };
 
   const onSelected = (d: number): boolean => {
-    return d === selectedDate?.getDate() && month === selectedDate?.getMonth();
+    return d === selectedDate?.getDate() && month === selectedDate?.getMonth() && year === selectedDate?.getFullYear();
```
Comment (hyochan) [inline packages/CalendarCarousel/index.tsx]:
```
const isSelectedDate = d === selectedDate?.getDate() && month === selectedDate?.getMonth() && year === selectedDate?.getFullYear();

return isSelectedDate;
```

File: packages/CalendarCarousel/index.tsx [hunk 6]
```diff
@@ -290,16 +272,30 @@ function CalendarCarousel<T>({
...
   };
 
   const isMarked = (d: number) : boolean => {
-    return markedDates.includes(d) && markedMonths.includes(month);
+    return markedDates.includes(d) && markedMonths.includes(month) && markedYears.includes(year);
```
Comment (hyochan) [inline packages/CalendarCarousel/index.tsx]:
```
const hasCurrentDateMarked = markedDates.includes(d) && markedMonths.includes(month) && markedYears.includes(year);

return hasCurrentDateMarked;
```

**Review by hyochan — CHANGES_REQUESTED** [review]
Please add a meaningful variable name for the readability of the conditional statement is bit complex.

**Comment by shljessie** [comment]
I can't do the isSelected and hasEvent changes you mentioned because I have to receive 'd' as a parameter! I did change the names of the functions for clarity though!

File: packages/CalendarCarousel/index.tsx [hunk 7]
```diff
@@ -115,73 +126,95 @@ const styles = StyleSheet.create<Style>({
...
       return onDateChanged?.(update);
     }
-
-    const update = new Date(
-      currentDate.getFullYear(),
-      currentDate.getMonth() + 1,
-      currentDate.getDate(),
-    );
-
+    const update = nextMonth;
```
Comment (hyochan) [inline packages/CalendarCarousel/index.tsx:204]: New line before / after paragraph code.

File: packages/CalendarCarousel/index.tsx [hunk 8]
```diff
@@ -115,73 +126,95 @@ const styles = StyleSheet.create<Style>({
...
-    const monthName = currentDate.toLocaleString('default', {
-      month: 'long',
-    });
-    const weekdays = [];
-    const year = currentDate.getFullYear();
-    const month = currentDate.getMonth();
+    const monthName = monthFormatter.format(currentDate);
 
-    const currentMonthDays = new Date(year, month + 1, 0).getDate();
+    const lastDate = new Date(year, month + 1, 0).getDate();
```
Comment (hyochan) [inline packages/CalendarCarousel/index.tsx]: No newline after the single line of code.

File: packages/CalendarCarousel/index.tsx [hunk 9]
```diff
@@ -192,81 +225,29 @@ function CalendarCarousel<T>({
       );
     }
 
-    const prevDays = [];
+    const prevDates = [];
     for (let idx = 0; idx < firstWeekday; idx++) {
```
Comment (hyochan) [inline packages/CalendarCarousel/index.tsx]: Newline before paragraph code.

File: packages/CalendarCarousel/index.tsx [hunk 10]
```diff
@@ -192,81 +225,29 @@ function CalendarCarousel<T>({
...
-          ),
-        )}>
-            <View style={styles.notActiveView}>
-              <Text style={styles.notActiveText}>{`${d}`}</Text>
-            </View>
-          </TouchableOpacity>,
-        );
-      }
+    const dates = [];
+    for (let idx = 1; idx <= lastDate; idx++) {
```
Comment (hyochan) [inline packages/CalendarCarousel/index.tsx]: Same here.

File: packages/CalendarCarousel/index.tsx [hunk 11]
```diff
@@ -192,81 +225,29 @@ function CalendarCarousel<T>({
...
+      dates.push(new Date(year, month, idx));
     }
 
-    const nextDays = [];
+    const nextDates = [];
     if (6 - lastWeekday >= 1) {
```
Comment (hyochan) [inline packages/CalendarCarousel/index.tsx]: Here!

File: packages/CalendarCarousel/index.tsx [hunk 12]
```diff
@@ -282,76 +263,150 @@ function CalendarCarousel<T>({
...
+      );
+    } else {
+      return (
+        <TouchableOpacity onPress={(): void => selectDate(setItemDay)}>
+          <View style={styles.notActiveView}>
+            <Text style={styles.notActiveText}>{`${itemDay}`}</Text>
+          </View>
+        </TouchableOpacity>
+      );
+    }
```
Comment (hyochan) [inline packages/CalendarCarousel/index.tsx]: You can remove `else` here and just return rest element.

File: packages/CalendarCarousel/index.tsx [hunk 13]
```diff
@@ -282,76 +263,150 @@ function CalendarCarousel<T>({
...
+      x: Math.floor(layoutWidth),
       animated: false,
     });
+  };
 
-    if (timeout) clearTimeout();
+  const scrollEffect = (e: NativeSyntheticEvent<NativeScrollEvent>) : void => {
+    const xValue = Math.floor(e.nativeEvent.contentOffset.x);
+    const maxLayoutFloor = Math.floor(layoutWidth) * 2;
+    if (!layoutWidth || layoutWidth === 1) return;
```
Comment (hyochan) [inline packages/CalendarCarousel/index.tsx]: Newline before / after `if` statement.

**Review by hyochan — CHANGES_REQUESTED** [review]
Much better @shljessie! Just a few more styling.

File: packages/CalendarCarousel/index.tsx [hunk 14]
```diff
@@ -282,76 +264,149 @@ function CalendarCarousel<T>({
...
+          data={calendarDates}
           numColumns={7}
-          data={calendarDays}
-          renderItem={({ item }): ReactElement => {
-            return item;
-          }}
+          renderItem ={({ item }): ReactElement => renderDates(item)}
           keyExtractor={(item, id): string => id.toString()}
         />
+        {renderEventText()}
```
Comment (hyochan) [inline packages/CalendarCarousel/index.tsx]: `renderEvent` since this is just and `element`!

**Comment by shljessie** [comment]
made changes! to the renderEvent()

**Comment by shljessie** [comment]
I found the function renderCalendars() that renders three Calendars to be unnecessary therefore, I created a function that erases that part.

**Comment by shljessie** [comment]
I think I'll start preparing for the presentation tomorrow based on the code I have now! let me know if there are any more changes I should make.

**Comment by hyochan** [comment]
image

Where did the `intl` go?

Also renderCalendar() is not using `currentDate` so all 3 calendars look the same

**Review by hyochan — APPROVED** [review]
Great work on this @shljessie :100:

**Pull request merged**